src/frontend/src/flows/authenticate/validateDerivationOrigin.ts

```diff
@@ -1,6 +1,7 @@
 import { Principal } from "@dfinity/principal";
 
 const ORIGIN_VALIDATION_REGEX = /^https:\/\/([\w-]+)(?:\.raw)?\.ic0\.app$/;
+const ALTERNATIVE_ORIGINS_LIMIT = 10;
```

Comment:
nit:

```suggestion
const MAX_ALTERNATIVE_ORIGINS = 10;
```

to clarify it's a maximum; also more consistent with other constants we have